transport/peer/peerlist/roundrobin.go

```diff
@@ -0,0 +1,181 @@
+package peerlist
+
+import (
+	"context"
+	"sync"
+
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/internal/errors"
+
+	"github.com/uber-go/atomic"
+)
+
+// NewRoundRobin creates a new round robin PeerList using
+func NewRoundRobin(peerIDs []transport.PeerIdentifier, agent transport.PeerAgent) (transport.PeerList, error) {
+	rr := &roundRobin{
+		peerToNode: make(map[string]*roundRobinNode, len(peerIDs)),
+		agent:      agent,
+		started:    atomic.NewBool(false),
+		nextNode:   nil,
+	}
+
+	err := rr.createPeers(peerIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	return rr, nil
+}
+
+type roundRobin struct {
+	sync.Mutex
+
+	peerToNode map[string]*roundRobinNode
+	agent      transport.PeerAgent
+	started    *atomic.Bool
+	nextNode   *roundRobinNode
```

Comment:
The "ring" package is what we’re looking for. Needs to be circular. It is the same story.